jwst/resample/tests/test_resample_step.py

```diff
@@ -865,6 +865,66 @@ def test_custom_refwcs_resample_imaging(nircam_rate, output_shape2, match,
     result.close()
 
 
+def test_custom_refwcs_pixel_shape_imaging(nircam_rate, tmp_path):
+
+    # make some data with a WCS and some random values
+    im = AssignWcsStep.call(nircam_rate, sip_approx=False)
+    rng = np.random.default_rng(seed=77)
+    im.data[:, :] = rng.random(im.data.shape)
+
+    crpix = (600, 550)
+    crval = (22.04, 11.98)
+    rotation = 15
+    ratio = 0.7
+
+    # first pass - create a reference output WCS:
+    result = ResampleStep.call(
+        im,
+        output_shape=(1205, 1100),
+        crpix=crpix,
+        crval=crval,
+        rotation=rotation,
+        pixel_scale_ratio=ratio
+    )
+
+    # make sure results are nontrivial
+    data1 = result.data
+    assert not np.all(np.isnan(data1))
+
+    # remove the bounding box so shape is set from pixel_shape
+    # and also set a top-level pixel area
+    pixel_area = 1e-13
+    refwcs = str(tmp_path / "resample_refwcs.asdf")
+    result.meta.wcs.bounding_box = None
+    asdf.AsdfFile({"wcs": result.meta.wcs,
+                   "pixel_area": pixel_area}).write_to(tmp_path / refwcs)
```

Comment:
This should be `.write_to(refwcs)`

Reply:
Yes it should!  Please go ahead and make that change.